File: 62.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        
        dp = [[0 for i in range(0,n)] for j in range(0,m)]
        
        for j in range(n):
            dp[0][j] = 1
            
        for j in range(m):
            dp[j][0] = 1
        
        
        for i in range(1,m):
            for j in range(1,n):
                dp[i][j] = dp[i-1][j] + dp[i][j-1]
        

        return dp[m-1][n-1]
        
    
    

File: 62.c


int uniquePaths(int m, int n){
    int dp[n][m];
    for(int i=0;i<n;i++){
        for(int j=0;j<m;j++){
            dp[i][j] = -1;
        }
    }
    
    for(int i=n-1;i>=0;i--){
        for(int j=m-1;j>=0;j--){
            if(i == (n-1) && j != (m-1)){
                dp[i][j] = 1;
            }
            else if(i != (n-1) && j == (m-1)){
                dp[i][j] = 1;
            }
            else if(i == n-1 && j == m-1){
                dp[i][j] = 1;
            }
            else{
                dp[i][j] = dp[i+1][j] + dp[i][j+1];
            }
        }
    }
    
    return dp[0][0];
}